tests/tracer/test_tracer.py

```diff
@@ -1526,3 +1529,12 @@ def test_get_report_hostname_default(get_hostname):
     child = spans[1]
     assert root.get_tag(HOSTNAME_KEY) is None
     assert child.get_tag(HOSTNAME_KEY) is None
+
+
+def test_service_mapping():
+    with override_env(dict(DD_SERVICE_MAPPING="foo:bar")):
```

Comment:
might be worth having a few more integration tests like this with different tags, multiple tags, etc

Reply:
Is there something in particular that I should be testing and that I'm overlooking? Part of the new functionality is also tested in `test_global_config`

Reply:
Oh yeah I was thinking that it'd be good to cover a few extra cases here as well since the internal implementation could change (eg. if we didn't use `config` in `start_span`) then it's possible that multiple tags could break without us noticing.

This is probably a bit overkill though but I figured it should be easy enough to add a few more test cases 😄

Reply:
@Kyle-Verhoog I've added a few extra test scenarios